rebasehelper/specfile.py

```diff
@@ -410,9 +389,13 @@ def write_updated_patches(self, patches, disable_inapplicable):
                     patch_inapplicable = [x for x in patches['inapplicable'] if patch_name in x]
                 else:
                     patch_inapplicable = None
-                if check_not_applied or patch_removed:
+                if check_not_applied:
                     self.spec_content[index] = '#{0} {1}\n'.format(' '.join(fields[:-1]),
                                                                    os.path.basename(patch_name))
```

Comment:
I think you are confusing `not applied` with `inapplicable` patches. AFAIK the former are the patches that have `PatchN` tag in the spec file but are lacking matching `%patchN` macro.
In my opinion these tags should be either removed or rather commented-out, but with a note justifying the reason of that.